aten/src/ATen/native/cuda/Distributions.cu

```diff
@@ -65,6 +65,43 @@ void poisson_cuda_kernel(
       });
 }
 
+template <typename scalar_t, typename accscalar_t>
+C10_DEVICE scalar_t binomial_cuda_kernel_worker(std::pair<uint64_t, uint64_t> seeds, scalar_t count, scalar_t prob) {
+  curandStatePhilox4_32_10_t state;
+  curandStatePhilox4_32_10_t* state_ptr = &state;
+  curand_init(
+      seeds.first,
+      blockIdx.x * blockDim.x + threadIdx.x,
```

Comment:
I don't think this is correct, because each thread can call the lambda multiple times. If implemented this way, it will generate the same random number multiple times. 

I don't think there is an easy way to do distribution kernels with `at::native::gpu_kernel` due to the reason I mentioned above. And my solution is to write my own elementwise kernels. See: https://github.com/pytorch/pytorch/pull/34004/files

Reply:
Thanks for the note, @zasdfgbnm. Unfortunately this won't explain the internal test failure (which is running CPU only)